cvapipe_analysis/tests/test_shape_space.py

```diff
@@ -0,0 +1,166 @@
+import filecmp
+from pathlib import Path
+import shutil
+import numpy as np
+import pandas as pd
+from cvapipe_analysis.tools import controller, shapespace
+from cvapipe_analysis.steps.shapemode.shapemode_tools import ShapeModeCalculator
+
+config = {
+    "aggregation": {"type": ["avg"]},
+    #"appName": "cvapipe_analysis",
+    "data": {
+        "nucleus": {"alias": "NUC", "channel": "dna_segmentation", "color": "#3AADA7"}
+    },
+    "distribute": {},
+    "features": {
+        "SHE": {
+            "aliases": ["NUC"],
+            "alignment": {"align": True, "reference": "nucleus", "unique": False},
+            "lmax": 16,
+            "sigma": 2,
+        },
+        "aliases": ["NUC"],
+    },
+    "parameterization": {
+        "inner": "NUC",
+        "number_of_interpolating_points": 32,
+        "outer": "MEM",
+        "parameterize": ["RAWSTR", "STR"],
+    },
+    "preprocessing": {
+        "filtering": {"csv": "", "filter": False, "specs": {}},
+        "remove_mitotics": True,
+        "remove_outliers": True,
+    },
+    "project": {
+        "local_staging": Path(__file__).parent / "all/shape_analysis/shape_space",
+        "overwrite": True,
+    },
+    "shapespace": {
+        "aliases": ["NUC"],
+        "map_points": [-2.0, -1.5, -1.0, -0.5, 0.0, 0.5, 1.0, 1.5, 2.0],
+        "number_of_shape_modes": 8,
+        "plot": {
+            "frame": False,
+            "limits": [-150, 150, -80, 80],
+            "swapxy_on_zproj": False,
+        },
+        "removal_pct": 0.25,
+        "sorter": "NUC",
+    },
+    "structures": {
+        "lamin": [
+            "nuclear envelope",
+            "#084AE7",
+            "{'raw': (475,1700), 'seg': (0,30), 'avgseg': (0,60)}",
+        ]
+    },
+}
+
+
+def random_shcoeffs_dataframe(nrows=100):
+    lmax = config["features"]["SHE"]["lmax"]
+    shcoeffs_cols = [
+        f"NUC_shcoeffs_L{L}M{m}{suffix}"
+        for L in range(1, lmax+1)
+        for m in range(0, L+1)
+        for suffix in ["C", "S"]
+        if suffix == "C" or m >= 1  # sine components are zero for m=0
+    ]
+    df = pd.DataFrame(
+        data=np.random.normal(size=(nrows, len(shcoeffs_cols))),
+        columns=shcoeffs_cols
+    )
+    df["NUC_shcoeffs_L0M0C"] = 5.0  # Each row is mostly spherical
+    df["NUC_shape_volume"] = np.random.normal(size=nrows)
+    df["NUC_position_depth"] = np.random.normal(size=nrows)
+    df["structure_name"] = "lamin"
+    return df
+
+
+def assert_NUC_PC_columns_nonzero(df):
```

Comment:
All other test methods have, Arrange, Act, and Assert.

Reply:
Not a test, just a helper function